backend/de.metas.business/src/main/sql/postgresql/ddl/functions/Reset_System_Base_Language.sql

```
@@ -0,0 +1,82 @@
+    SELECT l.ad_language INTO destination_language FROM ad_language l WHERE l.ad_language_id = p_ad_language_id;
+
+    IF (source_language IS NULL OR TRIM(source_language) = '') THEN
+        RAISE WARNING 'No source language is provided';
+        RETURN;
+    END IF;
+
+    IF (destination_language IS NULL OR TRIM(destination_language) = '') THEN
+        RAISE WARNING 'No destination language is provided';
+        RETURN;
+    END IF;
+
+    -- update languages
+    UPDATE ad_language
+    SET isbaselanguage='N', issystemlanguage='Y'
+    WHERE ad_language = source_language;
+
+    UPDATE ad_language
+    SET isbaselanguage = 'Y', issystemlanguage = 'Y'
+    WHERE ad_language = destination_language;
+
+    -- Iterate over Trls tables
+    FOR table_name IN SELECT UPPER(tablename) FROM ad_table WHERE UPPER(tablename) LIKE UPPER('%_Trl') ORDER BY 1
+        LOOP
+            base_table := SUBSTRING(table_name, 1, LENGTH(table_name) - 4);
+            BEGIN
+                -- Reset query output
+                query_set_template := '';
+                FOR v_columnname IN SELECT columnname FROM ad_column WHERE istranslated = 'Y' AND ad_table_id = get_table_id(base_table)
+                    LOOP
+                        query_set_template := query_set_template || FORMAT('%s = x.%s ,', v_columnname, v_columnname);
+                    END LOOP;
+
+                query_set_template := SUBSTRING(query_set_template, 1, LENGTH(query_set_template) - 1);
+
+                FOR base_table_id IN EXECUTE 'SELECT ' || base_table || '_ID FROM ' || base_table
+                    LOOP
+                        BEGIN
+                            final_update_query := FORMAT(REPLACE(update_query_template, 'base_table', base_table), base_table_id::varchar, destination_language, query_set_template);
+                            EXECUTE final_update_query;
+                        EXCEPTION
```

imho we don't have to handle the exception here and accept something which is partially OK.
IMHO we shall let it fail...